### PR TITLE
Update How It Works page

### PR DESCRIPTION
**Pull request by sshahcodes**
This PR:
* Assumes that "How it works" page should be the first page new Teleport user should read before diving into the docs pages. 
* Focus on adding/editing content that is more relevant to explain "how" Teleport feature works.
* Removes explanations related to "Why" or "What" part of Teleport features. They can be (and should be) covered in the product page.
* Makes consistent use of names such as  **Teleport Proxy Service**, **Teleport Auth Service** consistent across all how it works pages.
* Makes consistent use of sub headings such as **The Basics**, **Architecture**, **User Experience** across all how it works pages.
* Improve readability flow of the content that is already present in the page. 

Preview: https://teleport-docs-next-git-sshahcodes-update-h-4207e4-gravitational.vercel.app/how-it-works/
Page this PR will update: https://goteleport.com/how-it-works/

### CODE REVIEW

File: pages/how-it-works/index.mdx [hunk 1]
```diff
@@ -64,35 +46,37 @@ To create a minimal Teleport cluster, you must launch three services:
   </Box>
 </Flex>
 
+<br />
+Clients must authenticate with Teleport and receive a client certificate which automatically works for all resources in a cluster, i.e., after authentication, `ssh`, `kubectl`, `psql`, `mysql` and other remote access commands will be configured with the user's identity. Teleport offers a built-in database for user management, but for production use, we recommend integrating it with enterprise SSO based on Okta, Github, Google Apps, Active Directory, and other identity providers.
```
Comment (benarent) [inline pages/how-it-works/index.mdx:50]: I think this good, I wonder if we should provide internal links to our SSO Docs pages.

File: pages/how-it-works/index.mdx [hunk 2]
```diff
@@ -64,35 +46,37 @@ To create a minimal Teleport cluster, you must launch three services:
...
 from one organization to access designated servers inside of another
-organization’s cloud or on-premise environment.
+organization's cloud or on-premise environment.
 
 ## User Experience
 
-Users of [Teleport Desktop](https://goteleport.com/desktop-access/) access simply 
-use a browser to access Windows hosts.  
+Teleport users of [Teleport Desktop](https://goteleport.com/desktop-access/) access simply
+use a web browser to access Windows hosts.
```
Comment (benarent) [inline pages/how-it-works/index.mdx:72]: I wonder if should add something like 'RDP in the browser' to hit the right keywords

File: pages/how-it-works/index.mdx [hunk 3]
```diff
@@ -64,35 +46,37 @@ To create a minimal Teleport cluster, you must launch three services:
...
-web browser to authenticate and exits upon successful authentication. This configures 
+Teleport users of SSH, Kubernetes, and other command line-based clients, must
+authenticate first by executing the `tsh login` command. This command opens a
+web browser to authenticate and exits upon successful authentication. This configures
 users' CLI environment with short-lived certificates for access.
 
-After that, users will be able to access their SSH servers, Kubernetes clusters,
-web applications or databases. Teleport is backward compliant with existing
+After that, users will be able to access their SSH servers, Windows servers, Kubernetes clusters, web applications or databases. Teleport is backward compliant with existing
 client tools so users can continue to use `ssh`, `psql`, `mysql` or `kubectl`
```
Comment (benarent) [inline pages/how-it-works/index.mdx:80]: nit: We also support redis. So we may want to say, these main clients tools + others.  We also support GUI applications for Databases.

File: pages/how-it-works/index.mdx [hunk 4]
```diff
@@ -115,7 +99,7 @@ $ psql
 
 Teleport proxy serves the login screen on `https://proxy.example.com:3080`
 where users are asked for their username, password, and a 2nd factor. If a 3rd
-party identity such as Github is used, the proxy forwards the user to Github
+party identity provider such as Github is used, the proxy forwards the user to Github
```
Comment (benarent) [inline pages/how-it-works/index.mdx:102]:
```suggestion
party identity provider such as GitHub is used, the proxy forwards the user to GitHub
```

File: pages/how-it-works/index.mdx [hunk 5]
```diff
@@ -133,11 +117,11 @@ To learn more, check out the guide for the
...
 
-- _The event log_. It consists of well-documented JSON records of security
-  events. Examples of such events include login attempts, file transfers, code
-  execution, filesystem changes, or network activity.
-- _The recorded sessions_. All users’ interactive sessions to cluster nodes via
-  the ssh and kubectl commands are recorded for future replay.
+- **The event log:** Well-documented JSON records of security
+ events. Examples of such events include login attempts, file transfers, code
+ execution, filesystem changes, or network activity.
+- **The recorded sessions:** Recording of interactive sessions to SSH, RDP, Database, and kubectl commands. The recorded sessions can be replayed for audit purposes.
```
Comment (benarent) [inline pages/how-it-works/index.mdx:123]:
```suggestion
- **The recorded sessions:** Recording of interactive sessions to SSH, RDP and kubectl commands. The recorded sessions can be replayed for audit purposes.
```

File: pages/how-it-works/index.mdx [hunk 6]
```diff
@@ -133,11 +117,11 @@ To learn more, check out the guide for the
 The Teleport Auth service maintains the audit log of everything happening
 inside the environment. The audit log consists of two components:
 
-- _The event log_. It consists of well-documented JSON records of security
-  events. Examples of such events include login attempts, file transfers, code
-  execution, filesystem changes, or network activity.
-- _The recorded sessions_. All users’ interactive sessions to cluster nodes via
-  the ssh and kubectl commands are recorded for future replay.
+- **The event log:** Well-documented JSON records of security
```
Comment (benarent) [inline pages/how-it-works/index.mdx:120]:
```suggestion
- **The audit log:** Well-documented JSON records of security
```

**Review by benarent — COMMENTED** [review]
A few small comments but it looks good.

File: pages/how-it-works/index.mdx (hunk 1 above)
Comment (sshahcodes) [reply pages/how-it-works/index.mdx:50]: Added a link to enterprise SSO.

File: pages/how-it-works/index.mdx [hunk 7]
```diff
@@ -5,8 +5,8 @@ h1: Basic Concepts
 title: "How Teleport Works: Basic Concepts"
 headerTitle: How Teleport Works
 headerDescription:
-  Teleport is a single binary which provides secure access to SSH or RDP hosts,  Kubernetes clusters, web apps, PostgreSQL and MySQL databases behind NAT. Teleport is trivial to setup as a Linux daemon or in a Kubernetes pod.
-description: How SSH access works? How Kubernetes access works? This guide to Teleport concepts and architecture for SSH access, Kubernetes access answers these questions.
+ Teleport provides secure access to SSH servers, Windows servers, Kubernetes clusters, Databases, and Web Applications. Teleport is trivial to setup as a Linux daemon or in a Kubernetes pod.
```
Comment (ardeche07) [inline pages/how-it-works/index.mdx:8]:
```suggestion
 Teleport provides secure access to SSH or Windows servers, Windows Desktops, Kubernetes clusters, Databases, and Web Applications. Teleport is trivial to setup as a Linux daemon or in a Kubernetes pod.
```

File: pages/how-it-works/index.mdx [hunk 8]
```diff
@@ -17,10 +17,7 @@ import StructureDiagram from "components/StructureDiagram";
 
 ## The Basics
 
-Teleport is an identity-aware, multi-protocol access proxy which 
-implements protocols such as SSH, RDP,  HTTPS, Kubernetes API, MySQL, 
-PostgreSQL and others. It is completely transparent to client-side tools 
-and designed to work with everything in today's DevOps ecosystem.
+Teleport is a certificate authority and an identity-aware, multi-protocol access proxy which implements protocols such as SSH, RDP,  HTTPS, Kubernetes API, MySQL, PostgreSQL, MongoDB, Redis, and others. It is completely transparent to client-side tools and designed to work with everything in today's DevOps ecosystem.
```
Comment (ardeche07) [inline pages/how-it-works/index.mdx:20]:
```suggestion
Teleport is a certificate authority and identity-aware, multi-protocol access proxy which implements protocols such as SSH, RDP, HTTPS, Kubernetes API, and a variety of SQL and NoSQL databases. It is completely transparent to client-side tools and designed to work with everything in today's DevOps ecosystem.
```

File: pages/how-it-works/index.mdx [hunk 9]
```diff
@@ -64,37 +46,38 @@ To create a minimal Teleport cluster, you must launch three services:
...
+
 ## How Teleport Cluster Works
 
 The concept of a cluster is the foundation of the Teleport security model.
 
 - Users and servers all must join the same cluster before access can be granted.
 - To join a cluster, both users and servers must authenticate and receive certificates.
 - The Teleport auth service is the CA of the cluster, which issues certificates
-  for all supported protocols.
+ for both the users and servers with all supported protocols.
```
Comment (ardeche07) [inline pages/how-it-works/index.mdx:59]:
```suggestion
 for both users and servers with all supported protocols.
```

File: pages/how-it-works/index.mdx [hunk 10]
```diff
@@ -64,37 +46,38 @@ To create a minimal Teleport cluster, you must launch three services:
...
+Teleport users of SSH, Kubernetes, and other command line-based clients, must
+authenticate first by executing the `tsh login` command. This command opens a
+web browser to authenticate and exits upon successful authentication. This configures
 users' CLI environment with short-lived certificates for access.
 
-After that, users will be able to access their SSH servers, Kubernetes clusters,
-web applications or databases. Teleport is backward compliant with existing
-client tools so users can continue to use `ssh`, `psql`, `mysql` or `kubectl`
-commands as usual:
+After that, users will be able to access their SSH servers, Windows servers, Kubernetes clusters, web applications, or databases. Teleport is backward compliant with existing client tools, so users can continue to use client tools including `ssh`, `psql`, `mysql`, `kubectl`, and others as usual:
```
Comment (ardeche07) [inline pages/how-it-works/index.mdx:79]:
```suggestion
After that, users will be able to access their SSH servers, Windows servers, Kubernetes clusters, web applications, Windows desktops or databases. Teleport is backward compliant with existing client tools, so users can continue to use client tools including `ssh`, `psql`, `mysql`, `kubectl`, and others as usual:
```

**Review by ardeche07 — APPROVED** [review]
This looks good to me Sakshyam. I made a few suggestions.

**Pull request merged**